src/Rector/Contrib/PHPUnit/SpecificMethod/AssertComparisonToSpecificMethodRector.php

```diff
@@ -106,8 +110,16 @@ public function changeOrderArguments(MethodCall $methodCallNode): void
         /** @var BinaryOp $expression */
         $expression = $oldArguments[0]->value;
 
-        $firstArgument = new Arg($expression->right);
-        $secondArgument = new Arg($expression->left);
+        if ($this->isConstantValue($expression->right)) {
+            $firstArgument = new Arg($expression->right);
+            $secondArgument = new Arg($expression->left);
+        } elseif ($this->isConstantValue($expression->left)) {
+            $firstArgument = new Arg($expression->left);
+            $secondArgument = new Arg($expression->right);
+        } else {
+            $firstArgument = new Arg($expression->right);
+            $secondArgument = new Arg($expression->left);
+        }
```

Comment:
Can't we drop one condition here?

Reply:
Witch one? I’m searching from constant value in both members of the comparison, than I’m assuming a not Yoda Comparison.

Reply:
If you move  elseif `($this->isConstantValue($expression->left))` to be first condition you can drop elseif.

Reply:
```php 
if ($this->isConstantValue($expression->left)) {
 $firstArgument = new Arg($expression->left);
 $secondArgument = new Arg($expression->right);
} else {
 $firstArgument = new Arg($expression->right);
 $secondArgument = new Arg($expression->left);
}
```

Reply:
I think this kind of issue can be automated maybe ECS could do that, what do you think @TomasVotruba ?

Reply:
@mssimi How would you check it in some simple way? Coding Standard is not ready for this, since it has to check flow of control.
It's rather taks for PHPStan

Reply:
Simple workflow I have in my mind.

Consider code like this.

If
Code 1
elseif
Code 2
else
Code3

If Code 1 === Code3 then drop first condition and change second from elseif to if. ESC can't do this?

Reply:
It is possible, just from my point of view not worth the work.